client/src/components/forms/ProposalForm.jsx

```diff
@@ -4,58 +4,72 @@ import MonacoEditor from "@monaco-editor/react";
 import AlgorithmPreview from "../../pages/AlgorithmPreview"; // Assuming this path is correct
 import clsx from "clsx"; // For conditional class joining
 import { useSelector } from "react-redux";
+import isEqual from "lodash/isEqual";
+
+const defaultFormData = {
+  title: "",
+  problemStatement: "",
+  intuition: "",
+  explanation: "",
+  complexity: { time: "", space: "" },
+  difficulty: "Easy",
+  category: [],
+  tags: [],
+  links: [],
+  codes: [{ language: "", code: "" }],
+};
 
```

Comment:
When i click on preview what changes i have done in field user should able to see it but currently it's not working.